src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -286,7 +287,13 @@ protected function loadFile($file)
             $this->yamlParser = new YamlParser();
         }
 
-        return $this->validate($this->yamlParser->parse(file_get_contents($file)), $file);
+        try {
+            $configuration = $this->yamlParser->parse(file_get_contents($file));
+        } catch (ParseException $e) {
+            throw new \InvalidArgumentException(sprintf('The file "%s" does not contain valid YAML.', $file), 0, $e);
```

Comment:
this should probably use the exception of the component instead (as done when the file does not exist)